pkg/services/sqlstore/migrations/ualert/silences.go

```diff
@@ -0,0 +1,107 @@
+package ualert
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/matttproud/golang_protobuf_extensions/pbutil"
+	pb "github.com/prometheus/alertmanager/silence/silencepb"
+
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+)
+
+func (m *migration) addSilence(da dashAlert, rule *alertRule) error {
+	if da.State != "paused" {
+		return nil
+	}
+
+	uid, err := uuid.NewV4()
+	if err != nil {
+		return errors.New("failed to create uuid for silence")
+	}
+
+	n, v := getLabelForRouteMatching(rule.Uid)
+	s := &pb.MeshSilence{
+		Silence: &pb.Silence{
+			Id: uid.String(),
+			Matchers: []*pb.Matcher{
+				{
+					Type:    pb.Matcher_EQUAL,
+					Name:    n,
+					Pattern: v,
+				},
+			},
+			StartsAt:  time.Now(),
+			EndsAt:    time.Now().Add(365 * 20 * time.Hour), // 1 year.
+			CreatedBy: "Grafana Migration",
+			Comment:   "Created during auto migration to unified alerting",
+		},
+		ExpiresAt: time.Now().Add(365 * 20 * time.Hour), // 1 year.
+	}
+
+	m.silences = append(m.silences, s)
+	return nil
+}
+
+func (m *migration) writeSilencesFile() error {
+	var buf bytes.Buffer
+	for _, e := range m.silences {
+		if _, err := pbutil.WriteDelimited(&buf, e); err != nil {
+			return err
+		}
+	}
+
+	f, err := openReplace(silencesFileName(m.mg))
```

Comment:
Should we create the file if there are no silences?

Reply:
Should not matter because alertmanager would end up creating anyway. So it should be fine to create an empty file here.